iceoryx_hoofs/test/moduletests/test_posix_mutex.cpp

```diff
@@ -46,40 +46,62 @@ class Mutex_test : public Test
         }
     }
 
-    iox::posix::mutex sutNonRecursive{false};
-    iox::posix::mutex sutRecursive{true};
+    void signalThreadReady()
+    {
+        m_waitForThread.store(false);
+    }
+
+    void waitForThread()
+    {
+        while (m_waitForThread.load())
```

Comment:
same here ... and could you add `yield` to give the other thread a chance to run?